framework/src/postprocessors/ParsedPostprocessor.C

```diff
@@ -0,0 +1,105 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ParsedPostprocessor.h"
+
+registerMooseObject("MooseApp", ParsedPostprocessor);
+
+InputParameters
+ParsedPostprocessor::validParams()
+{
+  InputParameters params = GeneralPostprocessor::validParams();
+  params += FunctionParserUtils<false>::validParams();
+
+  params.addRequiredCustomTypeParam<std::string>(
+      "function", "FunctionExpression", "function expression");
+  params.addRequiredParam<std::vector<PostprocessorName>>("pp_names", "Post-processors arguments");
+  params.addParam<std::vector<std::string>>(
+      "constant_names", "Vector of constants used in the parsed function (use this for kB etc.)");
+  params.addParam<std::vector<std::string>>(
+      "constant_expressions",
+      "Vector of values for the constants in constant_names (can be an FParser expression)");
+  params.addParam<bool>(
+      "use_t", false, "Make time (t) variables available in the function expression.");
+
+  params.addClassDescription("Computes a parsed expression with post-processors");
+  return params;
+}
+
+ParsedPostprocessor::ParsedPostprocessor(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters),
+    FunctionParserUtils(parameters),
+    _n_pp(coupledPostprocessors("pp_names")),
+    _pp_names(getParam<std::vector<PostprocessorName>>("pp_names")),
+    _function(getParam<std::string>("function")),
+    _use_t(getParam<bool>("use_t"))
+{
+  // build postprocessors argument
+  std::string postprocessors;
+
+  // coupled  postprocessors
+  for (std::size_t i = 0; i < _n_pp; ++i)
+    postprocessors += (i == 0 ? "" : ",") + _pp_names[i];
+
+  // add time if required
+  if (_use_t)
+    postprocessors += (postprocessors.empty() ? "" : ",") + std::string("t");
+
+  // base function object
+  _func_F = std::make_shared<SymFunction>();
+
+  // set FParser internal feature flags
+  setParserFeatureFlags(_func_F);
+
+  // add the constant expressions
+  addFParserConstants(_func_F,
+                      getParam<std::vector<std::string>>("constant_names"),
+                      getParam<std::vector<std::string>>("constant_expressions"));
+
+  // parse function
+  if (_func_F->Parse(_function, postprocessors) >= 0)
+    mooseError(
+        "Invalid function\n", _function, "\nin ParsedAux ", name(), ".\n", _func_F->ErrorMsg());
```

Comment:
You can get rid of the ParsedAux name() stuff, you'll get a prefixed error with this info